bin/test_travis.sh

```diff
@@ -95,7 +95,11 @@ if [[ "${CC}" == *"clang"* ]] && [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
         export CXXFLAGS=""
     fi
 else
-    export CXXFLAGS="-Werror"
+    if [[ "${CC}" == *"gcc"* ]] && [[ "${BUILD_TYPE}" == "Debug" ]] && [[ "${BUILD_SHARED_LIBS}" == "yes" ]] && [[ "${WITH_GCC_6}" == "yes" ]] && [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
```

Comment:
Should we just use an env variable `USE_GLIBCXX_DEBUG` and set it in .travis.yml ?